### ISSUE TITLE
Question about embedders

### ISSUE THREAD

**Issue by vocdex** [levan92/deep_sort_realtime#34]
Thanks for sharing your code.
In original DeepSORT implementation, they use [Cosine Metric Learning](https://github.com/nwojke/cosine_metric_learning) for appearance feature embedding generation.
However, in this repo, you used mobilenet, torchreid and clip as embedder models. I guess these models are pre-trained on a general ImageNet dataset and therefore, provide more "general" embeddings. 
Have you compared the embedding (and/or tracking) results between cosine metric learning and the embedder models used here?
Does it make sense to use CLIP like embedder for specific objects such as a plant in a greenhouse or would it be better to train metric learning model?

**Comment by levan92** [comment]
The original DeepSORT embedder is trained on person re-identification datasets, meaning the embedder is specialised at differentiating human identities. Similarly, Torchreid models are also trained on person re-id datasets, but will likely be better than the original DeepSORT embedder due to recency and the fact that our selected model from Torchreid is trained in a domain generalised setting (aka multiple re-id datasets vs just 1). 

On the other hand, other embedders like mobilenet and CLIP in this repo will most likely be more generic if you do not want to look at humans. 

You will have to do your own experiments on your own specialised datasets if you want empirical results backing ur choice of embedder.
